bin/generate_test_data.rb

```diff
@@ -0,0 +1,170 @@
+    TEAM_NAMES = [
+      "bigdata",
+      "containers",
+      "dev",
+      "hardware",
+      "networking",
+      "ops",
+      "qa",
+      "security",
+      "sitereliability",
+      "ui"
+    ]
+
+    # Create some random users + the admin.
+    def create_users!
+      logins = Set.new
+
+      while logins.size < NUSERS
+        first_name  = FFaker::Name.first_name
+        last_name   = FFaker::Name.last_name
+        username    = "#{first_name[0]}#{last_name}".downcase
+
+        next if username =~ /\W/ || username == SPECIAL_USER
+        next if logins.include? username
+
+        User.create!(
+          username: username,
+          password: "12341234",
+          email:    "#{username}@portusexample.com"
+        )
+        logins << username
+      end
+
+      User.create!(
+        username: SPECIAL_USER,
+        password: "12341234",
+        email:    "admin@portusexample.com",
+        admin:    true
+      )
+    end
+
+    # Create the needed teams. Each team will have between 5 to 20 members and
+    # between 2 to 5 namespaces.
+    def create_teams_and_namespaces!
+      users = User.all
+
+      TEAM_NAMES.each do |team|
+        members = users.sample(rand(5..20))
+        t = Team.create!(
+          name:         team,
+          owners:       members.pop(rand(1..4)),
+          contributors: members.pop(rand(0..members.size)),
+          viewers:      members
+        )
+
+        rand(2..5).times do
+          name = FFaker::Food.ingredient.downcase.gsub(/[^#{User::USERNAME_CHARS}]/, "")
+          name += rand(0..1000).to_s
+
+          Namespace.create!(
+            name:     name,
+            team:     t,
+            public:   rand(0..1) == 1,
+            registry: Registry.get
+          )
+        end
+      end
+    end
+
+    # Push images to the registry.
+    def push_images!
+      hostname = Registry.get.hostname
+
+      `docker logout #{hostname}`
+      `docker pull library/alpine:latest`
+
+      Rails.logger.info "Pushing images. This could take a while..."
+
+      namespaces = Namespace.all.sample(rand(User.count..Namespace.count))
+      namespaces.each do |n|
+        user = n.global? ? User.find_by(username: "admin") : n.team.owners.first
+        name = user.username
+
+        s = system("docker login -u #{name} -e #{name}@portusexample.com -p 12341234 #{hostname}")
+        unless s
+          Rails.logger.error "Could not log in!"
+          exit 1
+        end
+
+        final = "#{hostname}/#{n.name}/alpine"
+        `docker tag alpine:latest #{final}`
+        `docker push #{final}`
+        `docker rmi #{final}`
+        `docker logout #{hostname}`
+      end
+    end
+  end
+end
+
+#
+# Initial checks.
+#
+
+if ARGV.size != 2
+  puts <<HERE
+usage: rails runner bin/generate_test_data.rb <hostname> <ssl>
+
+Where <hostname> is the hostname of your private registry, and <ssl> is a
+boolean that determines whether the registry is using SSL or not.
+HERE
```

Comment:
I would also add that docker daemon must be accessible from this machine (it has to run locally or you have to provide a `DOCKER_HOST` env variable)